### PR TITLE
Disable update for governance connectors in sub organizations

### PR DESCRIPTION
**Pull request by savindi7**
### Purpose
$subject

SMS OTP
Screenshot 2024-02-27 at 16 04 03

Email OTP
Screenshot 2024-02-27 at 16 04 11

Passkey
Screenshot 2024-02-27 at 16 04 19

### Related Issues
- https://github.com/wso2/product-is/issues/19848

### Related PRs
- Related PR `#1` or (None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/identity-providers/components/forms/authenticators/email-otp-authenticator-form.tsx [hunk 1]
```diff
@@ -158,14 +159,20 @@ export const EmailOTPAuthenticatorForm: FunctionComponent<EmailOTPAuthenticatorF
     } = props;
 
     const { t } = useTranslation();
+    const { isSubOrganization } = useGetCurrentOrganizationType();
 
     // This can be used when `meta` support is there.
     const [ , setFormFields ] = useState<EmailOTPAuthenticatorFormFieldsInterface>(undefined);
     const [ initialValues, setInitialValues ] = useState<EmailOTPAuthenticatorFormInitialValuesInterface>(undefined);
+    const [ isReadOnly, setIsReadOnly ] = useState<boolean>(undefined);
```
Comment (pavinduLakshan) [inline apps/console/src/features/identity-providers/components/forms/authenticators/email-otp-authenticator-form.tsx:167]:
```suggestion
    const [ isReadOnly, setIsReadOnly ] = useState<boolean>(!!isSubOrganization() || readOnly);
```

File: apps/console/src/features/identity-providers/components/forms/authenticators/email-otp-authenticator-form.tsx [hunk 2]
```diff
@@ -158,14 +159,20 @@ export const EmailOTPAuthenticatorForm: FunctionComponent<EmailOTPAuthenticatorF
...
 
     // SMS OTP length unit is set to digits or characters according to the state of this variable
     const [ isOTPAlphanumeric, setIsOTPAlphanumeric ] = useState<boolean>();
 
+    useEffect(() => {
+        setIsReadOnly(!!isSubOrganization() || readOnly);
+    }, [ readOnly ]);
```
Comment (pavinduLakshan) [inline apps/console/src/features/identity-providers/components/forms/authenticators/email-otp-authenticator-form.tsx:174]:
```suggestion
```

File: apps/console/src/features/identity-providers/components/forms/authenticators/fido-authenticator-form.tsx [hunk 3]
```diff
@@ -58,12 +58,18 @@ export const FIDOAuthenticatorForm: FunctionComponent<FIDOAuthenticatorFormProps
...
     const [ initialValues, setInitialValues ] = useState<FIDOAuthenticatorFormInitialValuesInterface>(undefined);
     const [
         isPasskeyProgressiveEnrollmentEnabled,
         setIsPasskeyProgressiveEnrollmentEnabled
     ] = useState<boolean>(undefined);
+    const [ isReadOnly, setIsReadOnly ] = useState<boolean>(undefined);
+
+    useEffect(() => {
+        setIsReadOnly(!!isSubOrganization() || readOnly);
+    }, [ readOnly ]);
```
[the remark this answers is not in the document]
Comment (savindi7) [reply apps/console/src/features/identity-providers/components/forms/authenticators/fido-authenticator-form.tsx]: fixed https://github.com/wso2/identity-apps/pull/5614/commits/9e020cd37c9948183709a853d255fcb1d03f837a

File: apps/console/src/features/identity-providers/components/forms/authenticators/email-otp-authenticator-form.tsx (hunk 1 above)
Comment (savindi7) [reply apps/console/src/features/identity-providers/components/forms/authenticators/email-otp-authenticator-form.tsx]: fixed https://github.com/wso2/identity-apps/pull/5614/commits/9e020cd37c9948183709a853d255fcb1d03f837a

**Review by brionmario — APPROVED** [review]

**Pull request merged**